srt-protocol/src/packet/mod.rs

```diff
@@ -138,4 +139,53 @@ impl Debug for Packet {
     }
 }
 
-pub type ReceivePacketResult = Result<(Packet, SocketAddr), PacketParseError>;
+#[derive(Debug)]
+pub enum ReceivePacketError {
+    Parse(PacketParseError),
+    Io(io::Error),
+}
+
+impl From<io::Error> for ReceivePacketError {
+    fn from(error: io::Error) -> Self {
+        ReceivePacketError::Io(error)
+    }
+}
+
+impl From<PacketParseError> for ReceivePacketError {
+    fn from(error: PacketParseError) -> Self {
+        ReceivePacketError::Parse(error)
+    }
+}
+
+impl fmt::Display for ReceivePacketError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        <Self as fmt::Debug>::fmt(self, f)
+    }
+}
+
+impl std::error::Error for ReceivePacketError {
+    fn source(&self) -> Option<&(dyn std::error::Error + 'static)> {
+        if let ReceivePacketError::Io(e) = self {
+            Some(e)
+        } else {
+            None
+        }
+    }
+}
+
+// NOTE: Eq, PartialEq are only here to accommodate structural comparison in tests without losing
+//  the ability to surface network errors for logging purposes, improve on this as needed, but don't
+//  remove.
+impl Eq for ReceivePacketError {}
```

Comment:
assert_eq! automatically pretty prints out the two values, this is valuable in quickly debugging test failures, I will use matches! on occasion when constructing an exemplar value is impossible, tedious, or non-essential, I don't want to lose the general ergonomics of testing just because io::Error was not implemented correctly :)

Reply:
Works for me